tests/unit/test_views.py

```diff
@@ -346,3 +345,158 @@ def testNoAuthentication(self):
         path = '/oauth2callback'.format(
             frontend.Version.currentString)
         self.assertEqual(501, self.app.get(path).status_code)
+
+    def testGenotypePhenotypeSearchBadRequest(self):
+        """
+        If nothing is provided to search on,
+        the system should return `bad request`
+        """
+        request = protocol.SearchFeaturesRequest()
+        response = self.sendPostRequest('/genotypephenotype/search', request)
+        self.assertEqual(400, response.status_code)
+
+    def testGenotypePhenotypeSearchFeature(self):
+        """
+        Search for evidence on a genomic feature given feature name
+        """
+        # simple string regexp
+        request = protocol.SearchFeaturesRequest()
+
+        request.feature = "KIT *wild"
```

Comment:
These tests are great, but they are really data driven tests, right? This depends on the server having a specific data set which it serves and we can test against. This module avoids this, and makes sure that all the tests are data agnostic, and really is only intended to test the basic properties of the http frontend. So, I think the tests in this module should be changed to use a simulated backend (more on this later), and also just check the basic HTTP stuff like the rest of the tests here.